shared/bridges/cctp/cosmos_bridge_client.go

```diff
@@ -171,11 +171,37 @@ func (c *CosmosBridgeClient) GetTxResult(ctx context.Context, txHash string) (*b
 			return nil, nil, ErrTxResultNotFound{TxHash: txHash}
 		}
 		return nil, nil, err
-	} else if result.TxResult.Code != 0 {
-		return big.NewInt(result.TxResult.GasUsed), &TxFailure{fmt.Sprintf("tx failed with code: %d and log: %s", result.TxResult.Code, result.TxResult.Log)}, nil
 	}
 
-	return big.NewInt(result.TxResult.GasUsed), nil, nil
+	// parse tx fee event and use as the gas cost. we are using the fee event
+	// as the gas cost, technically this is not always true for all cosmos
+	// txns, however for all of the transactions that the solver will submit
+	// and get results of via this function, this should be true
+	var fee sdk.Coin
+outer:
+	for _, event := range result.TxResult.GetEvents() {
```

Comment:
in what cases is the fee event emitted and what cases is it not?

https://github.com/skip-mev/relayer/blob/06732c77ac9f95fc3ebc998c29df4b5701576e6a/shared/bridges/cctp/cosmos_bridge_client.go#L191-L204

Reply:
from what i understand that is emitted every time gas is paid, so it would be in all of our txs. i like th relayers way of doing the paring tho, im going to change to that